File: fenicia-web/src/types/auth/get-module-response.ts
import { ModuleType } from "../../enums/auth-enums";


export type GetModuleResponse = {
    id: string;
    name: string;
    type: ModuleType;
    description?: string;
    icon?: string;
    isActive: boolean;
    sortOrder: number;
    price: number;
};

File: Src/Libs/Fenicia.Common/DTOs/Auth/Module/GetModuleResponse.cs
using System.ComponentModel.DataAnnotations;
using Fenicia.Common.Enums.Auth;

namespace Fenicia.Common.DTOs.Auth.Module;

public sealed record GetModuleResponse(
    [Required] Guid Id,
    [Required] [MaxLength(200)] string Name,
    [Required] ModuleType Type,
    [MaxLength(200)] string? Description,
    [MaxLength(200)] string? Icon,
    bool IsActive,
    int SortOrder,
    decimal? Price);